src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs

```diff
@@ -1,20 +1,33 @@
 ﻿namespace ServiceInsight.ServiceControl
 {
     using Caliburn.Micro;
+    using ServiceInsight.Framework;
 
     public class ServiceControlConnectionProvider
     {
         IEventAggregator eventAggregator;
+		IRxServiceControl serviceControl;
 
-        public ServiceControlConnectionProvider(IEventAggregator eventAggregator)
+        public ServiceControlConnectionProvider(IEventAggregator eventAggregator, IRxServiceControl serviceControl)
         {
             this.eventAggregator = eventAggregator;
+			this.serviceControl = serviceControl;
         }
 
         public void ConnectTo(string url)
         {
             Url = url;
-            eventAggregator.PublishOnUIThread(new ServiceControlConnectionChanged());
+			
+			eventAggregator.PublishOnUIThread(new ServiceControlConnectionChanged());
```

Comment:
Wouldn't this raise the ConnectionChanged event before actually changing the connection (below)?